### ISSUE TITLE
[consistent-type-definitions] Add option to use interface with methods

### ISSUE THREAD

**Issue by btd1337** [typescript-eslint/typescript-eslint#3894]
It is possible to add the possibility of not validating this rule when the interface has functions ([default Interface style used in languages like Java](https://www.w3schools.com/java/java_interface.asp))?

E.G.:

```ts
"@typescript-eslint/consistent-type-definitions": ["error", { "type", => "allowInterfaceWithFunctions" <= }]
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

- Create one interface with methods (Java and C# style)

```ts
interface Test {
   bar(): string;
   foo(): number;
}
```

- Linter will inform you to use `Type`

- Run `eslint` with `--fix`

- Code will be updated to:

```ts
type Test = {
   bar(): string;
   foo(): number;
}
```

- Eslint will report error: `',' expected`

**Rule**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-definitions": ["error", "type"],
  }
}
```

### Before
<img width="455" alt="Captura de Tela 2021-09-16 às 22 37 59" src="https://user-images.githubusercontent.com/11943860/133710436-fd27392e-b2c9-43ea-a078-a638aff49bc6.png">

### After auto fix
<img width="511" alt="Captura de Tela 2021-09-16 às 22 37 37" src="https://user-images.githubusercontent.com/11943860/133710422-d516b889-2ba4-46d1-93be-0ce3f22c8c37.png">

**Expected Result**

Don't suggest `Type` for `Interfaces` that contain functions (like Java and C# Interface styles)

**Actual Result**

Suggests change, performs auto-fix, but displays error after it.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.31.1` |
| `@typescript-eslint/parser`        | `4.31.1` |
| `TypeScript`                       | `4.4.3` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `12.18.3` |

**Comment by bradzacher** [comment]
Terminating a type property with a semicolon is valid syntax whether or not it is a type declaration or an interface.

The bug here is not that.
The bug here is that the fix ignored the fact that the interface was `export default`. You cannot directly `export default` a type declaration like you can an interface.

Thus the fixer should have instead done something like
```ts
type T = {...};
export default T;
```

**Comment by btd1337** [comment]
@bradzacher Yes, the one you said is the bug.

But besides the bug, I would like to know about the possibility of having a property that ignores the request to change from interface to type, when the interface only has functions (Java/C# interface style).

E.G.:

```ts
"@typescript-eslint/consistent-type-definitions": ["error", { "type", => "allowInterfaceWithFunctions" <= }]
```

**Comment by bradzacher** [comment]
No chance.
There's no reason that the rule should ignore  them aside from a purely stylistic choice made by your codebase.
With these stylistic lint rules we don't seek to satisfy all use cases because we'll never be able to. Instead we seek to support a well known, common subset.

If this is a style you want to use in your codebase - we purposely have a very permissive licence to allow people to fork out code.
